lib/mix/test/mix/tasks/compile.erlang_test.exs

```diff
@@ -59,4 +59,30 @@ defmodule Mix.Tasks.Compile.ErlangTest do
       refute File.regular?("_build/dev/lib/sample/ebin/b.beam")
     end
   end
+
+  test "compilation purges the module" do
```

Comment:
This test could be easier if you simply define an Elixir module before calling Compile.Erlang. Something like:

```elixir
defmodule :purge_test do
  def version, do: :v1
end
```

This way you don't need to write to disk and compile multiple times.